src/libraries/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedCmsTests.cs

```diff
@@ -1831,5 +1801,108 @@ private static void CheckNoSignature(byte[] encoded, bool badOid=false)
+                    {
+                        useSigner(new CmsSigner(SubjectIdentifierType.SubjectKeyIdentifier, cert));
+                    }
+                });
+        }
+
+        [ConditionalFact(typeof(SlhDsa), nameof(SlhDsa.IsSupported))]
+        public static void ComputeSignature_SlhDsa_DefaultDigest()
+        {
+#if !NETFRAMEWORK
+            // Test signer with public certificate and private key
+            AssertSignerHasCorrectDefaultDigest(
+                useSigner =>
+                {
+                    using (X509Certificate2 cert = Certificates.SlhDsaSha2_128s_Ietf.GetCertificate())
+                    using (SlhDsa key = SlhDsa.ImportSlhDsaSecretKey(SlhDsaAlgorithm.SlhDsaSha2_128s, SlhDsaTestData.IetfSlhDsaSha2_128sPrivateKeyValue))
+                    {
+                        useSigner(new CmsSigner(SubjectIdentifierType.SubjectKeyIdentifier, cert, key));
+                    }
+                });
+
+            // Test signer with private certificate
+            AssertSignerHasCorrectDefaultDigest(
+                useSigner =>
+                {
+                    using (X509Certificate2 cert = Certificates.SlhDsaSha2_128s_Ietf.TryGetCertificateWithPrivateKey())
+                    {
+                        useSigner(new CmsSigner(SubjectIdentifierType.SubjectKeyIdentifier, cert, (SlhDsa?)null));
+                    }
+                });
+#endif
+
+            // Test signer with private certificate
+            AssertSignerHasCorrectDefaultDigest(
+                useSigner =>
+                {
+                    using (X509Certificate2 cert = Certificates.SlhDsaSha2_128s_Ietf.TryGetCertificateWithPrivateKey())
+                    {
+                        useSigner(new CmsSigner(SubjectIdentifierType.SubjectKeyIdentifier, cert));
+                    }
+                });
+        }
+
+        private static void AssertSignerHasCorrectDefaultDigest(Action<Action<CmsSigner>> test, bool doSign = true)
+        {
+            // DigestAlgorithm property on new signer has correct default value
+            test(static signer => { Assert.Equal(DefaultHashForPlatform.Value, signer.DigestAlgorithm.Value); });
+
+            if (doSign)
+            {
+                // Signer signs with correct digest value
+                test(
+                    static signer =>
+                    {
+                        byte[] message = "Hello World!"u8.ToArray();
+                        SignedCms cms = new SignedCms(new ContentInfo(message));
+
+                        cms.ComputeSignature(signer);
+
+                        Assert.Equal(DefaultHashForPlatform.Value, cms.SignerInfos[0].DigestAlgorithm.Value);
+
+                        // Assert.NoThrow
+                        cms.SignerInfos[0].CheckSignature(verifySignatureOnly: true);
+                    });
+            }
+        }
+
+        private static bool IsNetFramework471OrLower =>
```

Comment:
This is fine, but since it won't change for a running process, it might be better to do

```suggestion
        private static bool IsNetFramework471OrLower { get; } =
```

So NetFX doesn't need to parse the framework name over, and over, and over